### PR TITLE
Em 98 - As a user, I would like to produce a visit summary letter

### PR DESCRIPTION
**Pull request by matrix-drorgo**
@matrix-amiel - please see that there are also bugs that you didn't moved to fix in EM-116 .
also here in this branch are bugs EM-363 ,PC-1557 that are also ON PR.

### CODE REVIEW

File: src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js [hunk 1]
```diff
@@ -29,6 +32,17 @@ const StyledExaminationStatusesWithIcons = ({
...
+  const [docID, setDoc] = React.useState(-1);
+  const createLetter = async () => {
+    let docId = await createSummaryLetter({
+      encounter: encounterData,
+      patientId: patient.id,
+      currentUser,
+      facility,
+    });
+    setDoc(docId);
+    store.dispatch(setEncounterAndPatient(encounterData, patient));
```
Comment (matrix-amiel) [inline src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js:44]: setEncounterAndPatient function saves the id of the encounter and patient in the store of redux, right? this is not needed and can makes problems (if clicked on prev encounters) in the encounter sheet forms. we need only to open the file in new tab.
image
The behavior of setEncounterAndPatient needs to be only in cameraClick function.
Or I missing something...

File: src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js [hunk 2]
```diff
@@ -13,12 +13,15 @@ import { store } from 'index';
 import { connect } from 'react-redux';
 import List from '@material-ui/core/List';
 import openDocumentInANewWindow from 'Utils/Helpers/openDocumentInANewWindow';
+import { createSummaryLetter } from '../../../../Utils/Helpers/Letters/createSummaryLetter';
```
Comment (matrix-amiel) [inline src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js:16]: change to absulate path

**Review by matrix-amiel — CHANGES_REQUESTED** [review]
looks :100: !

File: src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js (hunk 1 above)
Comment (matrix-drorgo) [reply src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js:44]: I saw that two but The camera is not disabled and you can open a letter when clicking on it . The problem that occur to me 
as a user is that that a letter can be opened from other encounter but If I am going out for a bit when I come back to the terminal I cannot know which encounter was it . so I added a line to the code that when you are clicking the camera the box of the encounter changes also If you like I can take it out ..... and leave it so you can click everywhere you like and letter will be opened and nobody will know which encounter it was ......

File: src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js (hunk 2 above)
Comment (matrix-drorgo) [reply src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js:16]: changed ...

File: src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js (hunk 1 above)
Comment (matrix-drorgo) [reply src/Components/Generic/EncounterSheet/PatientBackground/StyledExaminationStatusesWithIcons.js]: I have removed the line   store.dispatch(setEncounterAndPatient(encounterData, patient)); FROM

  const createLetter = async () => {
    let docId = await createSummaryLetter({
      encounter: encounterData,
      patientId: patient.id,
      currentUser,
      facility,
    });
    setDoc(docId);
  };

**Review by matrix-amiel — APPROVED** [review]

**Pull request merged**